src/dex/cables/cables.ts

```diff
@@ -0,0 +1,1006 @@
+      const pairsKeys = Object.keys(priceMap);
+
+      if (!pairsKeys.includes(pairKey)) {
+        // Revert
+        isInputQuote = true;
+        pairKey = `${normalizedDestToken.symbol}/${normalizedSrcToken.symbol}`;
+        if (!pairsKeys.includes(pairKey)) {
+          return null;
+        }
+      }
+
+      /**
+       * Orderbook
+       */
+      const priceData = priceMap[pairKey];
+
+      let orderbook: any[] = [];
+      if (side === SwapSide.BUY) {
+        orderbook = priceData.asks;
+      } else {
+        orderbook = priceData.bids;
+      }
+      if (orderbook?.length === 0) {
+        throw new Error(`Empty orderbook for ${pairKey}`);
+      }
+
+      const prices = this.calculateOrderPrice(
+        amounts,
+        orderbook,
+        side === SwapSide.SELL ? srcToken : destToken,
+        side === SwapSide.SELL ? destToken : srcToken,
+        side === SwapSide.SELL ? isInputQuote : !isInputQuote,
+      );
+
+      const result = [
+        {
+          prices: prices,
+          unit: BigInt(normalizedDestToken.decimals),
+          exchange: this.dexKey,
+          gasCost: CABLES_GAS_COST,
+          poolAddresses: [this.mainnetRFQAddress],
+          data: {},
+        },
+      ];
+
+      return result;
+    } catch (e: unknown) {
+      this.logger.error(
+        `Error in getPricesVolume`,
+        {
+          srcToken: srcToken.address || srcToken.symbol,
+          destToken: destToken.address || destToken.symbol,
+          side,
+        },
+        e,
+      );
+      return null;
+    }
+  }
+
+  getCalldataGasCost(poolPrices: PoolPrices<CablesData>): number | number[] {
+    return (
+      CALLDATA_GAS_COST.DEX_OVERHEAD +
+      // addresses: makerAsset, takerAsset, maker, taker
+      CALLDATA_GAS_COST.ADDRESS * 4 +
+      // uint256: expiry
+      CALLDATA_GAS_COST.wordNonZeroBytes(16) +
+      // uint256: nonceAndMeta, makerAmount, takerAmount
+      CALLDATA_GAS_COST.AMOUNT * 3 +
+      // bytes: _signature (65 bytes)
+      CALLDATA_GAS_COST.FULL_WORD * 2 +
+      CALLDATA_GAS_COST.OFFSET_SMALL
+    );
+  }
+
+  async initializePricing(blockNumber: number): Promise<void> {
+    if (!this.dexHelper.config.isSlave) {
+      this.rateFetcher.start();
+    }
+    return;
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return null;
+  }
+
+  releaseResources?(): AsyncOrSync<void> {
+    if (this.rateFetcher) {
+      this.rateFetcher.stop();
+    }
+  }
+
+  normalizeAddress(address: string): string {
+    return address.toLowerCase() === ETHER_ADDRESS
+      ? NULL_ADDRESS
+      : address.toLowerCase();
+  }
+
+  async setTokensMap() {
+    const tokens = await this.getCachedTokens();
+
+    if (tokens) {
+      this.tokensMap = Object.keys(tokens).reduce((acc, key) => {
+        //@ts-ignore
+        acc[tokens[key].address.toLowerCase()] = tokens[key];
+        return acc;
+      }, {});
+    }
+  }
+
+  async getTopPoolsForToken(
```

Comment:
If this integration is not going to introduce new token pairs as we already have on other dexs, I prefer to just return empty array. Otherwise, we are going to too much multiple Redis, RPC, usdTokenPrice calls. So, imho, it should be either emptified or simplified.

Reply:
fixed